src/main/java/org/zeplinko/commons/lang/ext/core/AbstractResult.java

```diff
@@ -0,0 +1,129 @@
+package org.zeplinko.commons.lang.ext.core;
+
+import jakarta.annotation.Nonnull;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@SuppressWarnings("LombokGetterMayBeUsed")
+public abstract class AbstractResult<D, E> {
```

Comment:
I was thinking, the Abstract one can be similar to Either in swift. The abstract one can be `<Data, Data>`

Then a specialization of this can be `<Data, Error>`.

Thoughts?

Reply:
Yes, we will do that. We will have to create AbstractEither and Either to have that hierarchy. We do have an open issue for the Either class, let's discuss during that time.

But it gave me an idea to rename this class to not skew towards one of it child. I will rename this to AbstractOutcome.

ref: #43